tests/test_build.py

```diff
@@ -540,30 +540,41 @@ def test_edit_page_url(sphinx_build_factory, html_context, edit_url):
     assert edit_link[0].attrs["href"] == edit_url, f"edit link didn't match {edit_link}"
 
 
-def test_new_google_analytics_id(sphinx_build_factory):
-    confoverrides = {"html_theme_options.google_analytics_id": "G-XXXXX"}
-    sphinx_build = sphinx_build_factory("base", confoverrides=confoverrides)
-    sphinx_build.build()
-    index_html = sphinx_build.html_tree("index.html")
-
-    # Search all the scripts and make sure one of them has the Google tag in there
-    tags_found = False
-    for script in index_html.select("script"):
-        if script.string and "gtag" in script.string and "G-XXXXX" in script.string:
-            tags_found = True
-    assert tags_found is True
-
-
-def test_old_google_analytics_id(sphinx_build_factory):
-    confoverrides = {"html_theme_options.google_analytics_id": "UA-XXXXX"}
+@pytest.mark.parametrize(
+    "provider,tags",
+    [
+        # deprecated
```

Comment:
```suggestion
        # TODO: Deprecate old-style analytics config >= 0.12
```

Reply:
sorry for being nitpicky here but I find it much better to define a version in our deprecation comments to resolve ambiguity about when to delete stuff :-)